github/github.go

```diff
@@ -981,8 +981,8 @@ func (r *TwoFactorAuthError) Error() string { return (*ErrorResponse)(r).Error()
 // RateLimitError occurs when GitHub returns 403 Forbidden response with a rate limit
 // remaining value of 0.
 type RateLimitError struct {
-	Rate     Rate           // Rate specifies last known rate limit for the client
-	Response *http.Response // HTTP response that caused this error
+	Rate     Rate           `json:"rate"`    // Rate specifies last known rate limit for the client
+	Response *http.Response `json:"-"`       // HTTP response that caused this error
 	Message  string         `json:"message"` // error message
 }
```

Comment:
@gmlewis  Is it really needed to create "marshal" test for structures like this and the `AbuseRateLimitError`, or `Rate`?

As far as I understand it, it is not a structure that represents anything we send or receive from GitHub.

Reply:
Probably not then... let's just please be super-careful to try and imagine ways that modifying things like this might affect end-users of this client library.  If it breaks things, we need to think long and carefully about whether it is worth the change since we have thousands of users of this client library in the wild.

Reply:
I left the structures as they were and removed the two tests `TestRateLimitError_Marshal` and `TestRateLimitError_Marshal`, as I believe they do not provide much value.

What do you think about this idea?